webgpu.h

```diff
@@ -0,0 +1,1025 @@
+    uint32_t baseMipLevel;
+    uint32_t mipLevelCount;
+    uint32_t baseArrayLayer;
+    uint32_t arrayLayerCount;
+    wgpuTextureAspect aspect;
+} wgpuTextureViewDescriptor;
+
+typedef struct wgpuVertexAttributeDescriptor {
+    uint32_t shaderLocation;
+    uint64_t offset;
+    wgpuVertexFormat format;
+} wgpuVertexAttributeDescriptor;
+
+typedef struct wgpuBindGroupDescriptor {
+    const void* nextInChain;
+    wgpuStructureType structureType;
+    wgpuBindGroupLayout layout;
+    uint32_t bindingCount;
+    wgpuBindGroupBinding const * bindings;
+} wgpuBindGroupDescriptor;
+
+typedef struct wgpuBindGroupLayoutDescriptor {
+    const void* nextInChain;
+    wgpuStructureType structureType;
+    uint32_t bindingCount;
+    wgpuBindGroupLayoutBinding const * bindings;
+} wgpuBindGroupLayoutDescriptor;
+
+typedef struct wgpuColorStateDescriptor {
+    const void* nextInChain;
+    wgpuStructureType structureType;
+    wgpuTextureFormat format;
+    wgpuBlendDescriptor alphaBlend;
+    wgpuBlendDescriptor colorBlend;
+    wgpuColorWriteMask writeMask;
+} wgpuColorStateDescriptor;
+
+typedef struct wgpuComputePipelineDescriptor {
+    const void* nextInChain;
+    wgpuStructureType structureType;
+    wgpuPipelineLayout layout;
+    wgpuPipelineStageDescriptor const * computeStage;
+} wgpuComputePipelineDescriptor;
+
+typedef struct wgpuDepthStencilStateDescriptor {
+    const void* nextInChain;
+    wgpuStructureType structureType;
+    wgpuTextureFormat format;
+    bool depthWriteEnabled;
+    wgpuCompareFunction depthCompare;
+    wgpuStencilStateFaceDescriptor stencilFront;
+    wgpuStencilStateFaceDescriptor stencilBack;
+    uint32_t stencilReadMask;
+    uint32_t stencilWriteMask;
+} wgpuDepthStencilStateDescriptor;
+
+typedef struct wgpuRenderPassColorAttachmentDescriptor {
+    wgpuTextureView attachment;
+    wgpuTextureView resolveTarget;
+    wgpuLoadOp loadOp;
+    wgpuStoreOp storeOp;
+    wgpuColor clearColor;
+} wgpuRenderPassColorAttachmentDescriptor;
+
+typedef struct wgpuTextureCopyView {
+    const void* nextInChain;
+    wgpuStructureType structureType;
+    wgpuTexture texture;
+    uint32_t mipLevel;
+    uint32_t arrayLayer;
+    wgpuOrigin3D origin;
+} wgpuTextureCopyView;
+
+typedef struct wgpuTextureDescriptor {
+    const void* nextInChain;
+    wgpuStructureType structureType;
+    wgpuTextureUsageBit usage;
+    wgpuTextureDimension dimension;
+    wgpuExtent3D size;
+    uint32_t arrayLayerCount;
+    wgpuTextureFormat format;
+    uint32_t mipLevelCount;
+    uint32_t sampleCount;
+} wgpuTextureDescriptor;
+
+typedef struct wgpuVertexBufferDescriptor {
+    uint64_t stride;
+    wgpuInputStepMode stepMode;
+    uint32_t attributeCount;
+    wgpuVertexAttributeDescriptor const * attributes;
+} wgpuVertexBufferDescriptor;
+
+typedef struct wgpuRenderPassDescriptor {
+    uint32_t colorAttachmentCount;
+    const wgpuRenderPassColorAttachmentDescriptor* const * colorAttachments;
```

Comment:
why is this a pointer to a pointer, as opposed to just a pointer to the first element in an array?

Reply:
Because this descriptor needs to be extended and could be swapped for a differently typed descriptor (linked to [1])